xdsl/utils/hints.py

```diff
@@ -141,7 +141,9 @@ def get_type_var_mapping(
     # Get the generic parent
     orig_bases: Iterable[Any] = getattr(cls, "__orig_bases__")
     orig_bases = [
-        orig_base for orig_base in orig_bases if get_origin(orig_base) is not Generic
+        orig_base
```

Comment:
now that Mathieu has fixed this in #1794, please ignore changes in this file, I'll merge this PR only after that one